backend/helper_zero/views/search.py

```diff
@@ -43,40 +73,94 @@ def _filter_org_list_by_donation_requests(org_search_results, item_type):
-            lon__gte=br_lon,
-            is_dropoff_only__in=delivery_values,
-        )
+        org_query_set = _process_lat_lon_query(lat, lon, radius, org_filter_params, donation_request_filter_params)
     else:
         # If no location data, default to showing any orgs that have open requests, capping at 20
-        org_query_set = Organization.objects.filter(
-            donation_requests__isnull=False
-        ).order_by("name")[:20]
+        org_query_set = _process_default_query(org_filter_params, donation_request_filter_params)
 
     org_search_results = OrganizationSerializer(org_query_set, many=True).data
     # Remove duplicates from query set.
     # This appears to be a known django issue with sqlite and filtering on models with FKs.
     org_search_results = [i for n, i in enumerate(org_search_results) if i not in org_search_results[n + 1:]]
 
     if lat and lon:
-        # Order by closest to search location
+        # Order by closest to search location before pagination
         org_search_results.sort(key=lambda x: _get_distance(float(x['lat']), lat, float(x['lon']), lon))
 
     org_search_results = _paginate_results(org_search_results, offset)
     return org_search_results
 
 
+def _process_zipcode_query(zipcode, org_filter_params, donation_request_filter_params):
+    if donation_request_filter_params.get('item_type'):
+        org_query_set = Organization.objects.filter(
+            zipcode=zipcode,
+            donation_requests__item_type=donation_request_filter_params['item_type'],
+            **org_filter_params,
+        )
+    else:
+        org_query_set = Organization.objects.filter(
+            zipcode=zipcode,
+            **org_filter_params,
+        )
+    return org_query_set
+
+
+def _process_lat_lon_query(lat, lon, radius, org_filter_params, donation_request_filter_params):
+    tl_lon, tl_lat, br_lon, br_lat = get_search_bounding_box(
+        lat,
+        lon,
+        radius,
+    )
+    if donation_request_filter_params.get('item_type'):
+        org_query_set = Organization.objects.filter(
+            lat__lte=tl_lat,
+            lat__gte=br_lat,
+            lon__lte=tl_lon,
+            lon__gte=br_lon,
+            donation_requests__item_type=donation_request_filter_params['item_type'],
+            **org_filter_params
+        )
+    else:
+        org_query_set = Organization.objects.filter(
+            lat__lte=tl_lat,
+            lat__gte=br_lat,
+            lon__lte=tl_lon,
+            lon__gte=br_lon,
+            **org_filter_params
+        )
+    return org_query_set
+
+
+def _process_default_query(org_filter_params, donation_request_filter_params):
+    if donation_request_filter_params.get('item_type'):
+        org_query_set = Organization.objects.filter(
+            donation_requests__isnull=False,
+            donation_requests__item_type=donation_request_filter_params['item_type'],
+            **org_filter_params
+        ).order_by("name")[:20]
```

Comment:
NIT: Probably throw the magic number 20 into a constant so we can more easily change the number of results displayed later if we think it looks better with a different number

Reply:
good idea (y) I realized we don't have to limit to 20 results here in our query, and should be letting the pagination handle it so changed that as well